core/frontend/src/libs/console-logger.ts

```diff
@@ -0,0 +1,161 @@
+  stack?: string
+  args?: any[]
+}
+
+class ConsoleLogger {
+  private session: Session | null = null
+
+  readonly originalConsole: {
+    log: typeof console.log
+    info: typeof console.info
+    warn: typeof console.warn
+    error: typeof console.error
+    debug: typeof console.debug
+  }
+
+  constructor() {
+    this.originalConsole = {
+      log: console.log,
+      info: console.info,
+      warn: console.warn,
+      error: console.error,
+      debug: console.debug,
+    }
+  }
+
+  async initialize(): Promise<void> {
+    if (this.session !== null) {
+      return
+    }
+
+    try {
+      const protocol = window.location.protocol === 'https:' ? 'wss' : 'ws'
+      const url = `${protocol}://${window.location.host}/zenoh-api/`
+      const config = new Config(url)
+      this.session = await Session.open(config)
+
+      this.interceptConsole()
+      this.interceptWindow()
+    } catch (error) {
+      console.error('[ConsoleLogger] Failed to initialize:', error)
+    }
+  }
+
+  private readonly onError = (event: ErrorEvent): void => {
+    this.publishMessage('error', [event.message, event.filename, event.lineno, event.colno, event.error])
+  }
+
+  private readonly onUnhandledRejection = (event: PromiseRejectionEvent): void => {
+    this.publishMessage('error', [event.reason, event.promise, event.type])
+  }
+
+  private interceptWindow(): void {
+    window.addEventListener('error', this.onError)
+    window.addEventListener('unhandledrejection', this.onUnhandledRejection)
+  }
+
+  private interceptConsole(): void {
+    console.log = (...args: any[]) => {
+      this.originalConsole.log(...args)
+      this.publishMessage('log', args)
+    }
+
+    console.info = (...args: any[]) => {
+      this.originalConsole.info(...args)
+      this.publishMessage('info', args)
+    }
+
+    console.warn = (...args: any[]) => {
+      this.originalConsole.warn(...args)
+      this.publishMessage('warn', args)
+    }
+
+    console.error = (...args: any[]) => {
+      this.originalConsole.error(...args)
+      this.publishMessage('error', args)
+    }
+
+    console.debug = (...args: any[]) => {
+      this.originalConsole.debug(...args)
+      this.publishMessage('debug', args)
+    }
+  }
+
+  private publishMessage(level: ConsoleMessage['level'], args: any[]): void {
+    if (!this.session) {
+      return
+    }
+
+    try {
+      const clonedArgs = [...args]
+
+      const message: ConsoleMessage = {
+        level,
+        message: clonedArgs.map((arg) => ConsoleLogger.stringifyArgument(arg)).join(' '),
+        frontend_id: frontend.frontend_id,
+        args: clonedArgs.length > 1 ? clonedArgs : undefined,
+      }
+
+      if (level === 'error' && clonedArgs[0] instanceof Error) {
+        message.stack = clonedArgs[0].stack
+      }
+
+      const topic = `frontend/${frontend.frontend_id}`
+      const payload = JSON.stringify(message)
+
+      this.session.put(topic, payload, { encoding: Encoding.APPLICATION_JSON })
```

Comment:
I would recommend using https://docs.foxglove.dev/docs/visualization/message-schemas/log
and using as schema for APPLICATION_JSON `Foxglove.Log`